gotrue/_async/client.py

```diff
@@ -0,0 +1,630 @@
+from functools import partial
+from json import dumps, loads
+from re import match
+from threading import Timer
+from time import time
+from typing import Any, Callable, Dict, Optional, Union, cast
+from urllib.parse import parse_qs, urlparse
+from uuid import uuid4
+
+from ..constants import COOKIE_OPTIONS, DEFAULT_HEADERS, GOTRUE_URL, STORAGE_KEY
+from ..types import (
+    ApiError,
+    AuthChangeEvent,
+    CookieOptions,
+    Provider,
+    Session,
+    Subscription,
+    User,
+    UserAttributes,
+)
+from .api import AsyncGoTrueApi
+from .storage import AsyncMemoryStorage, AsyncSupportedStorage
+
+HTTPRegexp = "/^http://"
+
+
+class AsyncGoTrueClient:
+    def __init__(
+        self,
+        *,
+        url: str = GOTRUE_URL,
+        headers: Dict[str, str] = DEFAULT_HEADERS,
+        auto_refresh_token: bool = True,
+        persist_session: bool = True,
+        local_storage: AsyncSupportedStorage = AsyncMemoryStorage(),
+        cookie_options: CookieOptions = CookieOptions.from_dict(COOKIE_OPTIONS),
+    ) -> None:
+        """Create a new client
+
+        url : str
+            The URL of the GoTrue server.
+        headers : Dict[str, str]
+            Any additional headers to send to the GoTrue server.
+        auto_refresh_token : bool
+            Set to "true" if you want to automatically refresh the token before
+            expiring.
+        persist_session : bool
+            Set to "true" if you want to automatically save the user session
+            into local storage.
+        local_storage : SupportedStorage
+            The storage engine to use for persisting the session.
+        cookie_options : CookieOptions
+            The options for the cookie.
+        """
+        if match(HTTPRegexp, url):
```

Comment:
/nit Perhaps we can consider just using a `.startswith('http')` ? Using a regexp for this seems a little heavy.

Reply:
Ok 👍🏼